LDM-633.tex

```diff
@@ -0,0 +1,319 @@
+}
+
+% Change history defined here.
+% Order: oldest first.
+% Fields: VERSION, DATE, DESCRIPTION, OWNER NAME.
+% See LPM-51 for version number policy.
+\setDocChangeRecord{%
+  \addtohist{1}{YYY-MM-DD}{Unreleased.}{Mikolaj Kowalik}
+}
+
+\begin{document}
+
+% Create the title page.
+% Table of contents is added automatically with the "toc" class option.
+\maketitle
+
+\section{Introduction}
+
+This document describes use cases for the LSST Batch Production Service.  These
+use cases are driven by production offline processing, but may also apply to
+other user needs, e.g. pipeline developers.
+
+\section{Definitions}
+
+\begin{description}
+  \item[Pipeline] 
+    An ordered sequence of individual steps.
+  \item[Payload]
+    A sequence of pipelines that performs particular data analysis and product
+    generation.
+  \item[Campaign]
+    A set of all pipelines executions needed to achieve a LSST objective.
+\end{description}
+
+\section{Actors}
+
+\begin{description}
+  \item[Operator (OP)]
+    A person responsible for offline processing (e.g. DRP).
+
+  \item[Campaign Manager (CM)]
+    A system or person managing processing campaigns.
+
+  \item[Worfklow Management System (WMS)]
+    A system managing an execution of a pipeline.
+\end{description}
+
+\section{Scope of Operations}
+An Operator may wish to perform operations, such as submissions, restarts,
+pausing, terminating, monitoring, etc., to single pipelines or groups of
+pipelines determined by some description, such as campaign, ones currently
+running, etc.
+
+\section{Configuration}
+
+\usecase{BPS}{Override default configuration options}
+The Operator wants to override a subset of configuration options for a
+campaign/pipeline at any configuration level (i.e., global, site, campaign,
+pipeline, job, SuperTask). These overrides include but are not limited to:
+\begin{itemize}
+  \item
+    execution configuration of a SuperTask (e.g., expected amount of memory
+    needed),
+  \item
+    science configuration of a SuperTask (e.g., update particular threshold),
+  \item
+    globally override certain configurations (e.g., whether to bring all output
+    datasets home from job including intermediates),
+  \item
+    specify to process campaign/pipeline/jobs on particular computational
+    platform(s),
+  \item
+    excluding certain data from a campaign,
+  \item
+    modify pipeline sequence of SuperTasks (e.g., only run half of a pipeline
+    for debugging).
+\end{itemize}
+
+\usecase{BPS}{Select specific software version(s)}
+The Operator requests to run a pipeline with specific versions of the LSST
+packages (e.g. wants to use a particular LSST Stack release or specific
+versions of individual packages).
+
+\usecase{BPS}{Order steps in a pipeline}
+The Operator specifies the order in which steps constituting the pipeline
+should be executed (e.g. to postpone execution of steps which are likely to
+fail towards the end of a pipeline or start processing from specific steps).
```

Comment:
In general I think it might be helpful to readers if there are some concrete examples, e.g. with some real science steps/pipelines.

Reply:
As we discussed, it's _not_ about altering the DAG representing a pipeline but specifying the order its nodes (pipeline steps) should be explored (executed). Though I agree that current wording may be misleading. I'll try to make clear.